gateway/src/apicast/policy/3scale_batcher/reports_batcher.lua

```diff
@@ -91,7 +91,7 @@ function _M:get_all(service_id)
 
     local report = keys_helper.report_from_key_batched_report(key, value)
 
-    if value and value > 0 and report.service_id == service_id then
+    if value and value > 0 and report and report.service_id == service_id then
```

Comment:
I think that the failing test was because of this: https://github.com/3scale/APIcast/blob/e825115209d53cf6c564e0aad889412eb5e758d4/t/apicast-policy-3scale-batcher.t#L613 We should change that to "azp".

Reply:
Yes, but still `report_from_key_batched_report` can return `nil` and this will crash. That should not happen.

Reply:
Yes, that too. I think that apart from that, we should add a warning in `report_from_key_batched_report`, because if it returns nil, it means that the policy received a combination of "auth params" (app_id, app_key, ...) that it's not supported.

Reply:
That probably deserves its own PR with a test. Will create one.

Reply:
https://github.com/3scale/APIcast/pull/1009